README.md

```diff
@@ -1,85 +1,135 @@
+`hg`, `svn`, and `bzr`.
 
-It supports many options, e.g. it can adjust resulting tarball
-parameters, include or exclude particular files when creating this
-tarball or generate an rpm changelog from the SCM commit log. For the
-full list of options please see tar_scm.service.in.
+`tar_scm` supports many options, e.g. it can adjust resulting tarball
+parameters, include or exclude particular files when creating the
+tarball, or generate an `rpm` changelog from the SCM commit log. For the
+full list of options please see `tar_scm.service.in`.
 
-Apart from various SCM like git, hg, bzr or svn it additionally
+Apart from various SCM like git, hg, bzr or svn, it additionally
 supports `--url` option that allows you to specify URL of the upstream
 tarball to be downloaded.
 
-This is the git repository for
-[openSUSE:Tools/obs-service-tar_scm](https://build.opensuse.org/package/show/openSUSE:Tools/obs-service-tar_scm).
-The authoritative source is
-https://github.com/openSUSE/obs-service-tar_scm
-
-The service can be used in combination with other services like
+`tar_scm` can be used in combination with other services like
 [download_files](https://github.com/openSUSE/obs-service-download_files),
 [extract_file](https://github.com/openSUSE/obs-service-extract_file),
 [recompress](https://github.com/openSUSE/obs-service-recompress) or
 [set_version](https://github.com/openSUSE/obs-service-set_version)
 e.g. within the [GIT integration](https://en.opensuse.org/openSUSE:Build_Service_Concept_SourceService#Example_2:_GIT_integration)
 workflow.
 
-## Archive Formats
+**`tar_scm` is deprecated in favour of `obs_scm`.**
 
-### obscpio
+### obs_scm
 
-The `obscpio` archive format is the new OBS specific archive format
-based on `cpio` which can be consumed and generated by the [OBS Delta
-Store](http://openbuildservice.org/help/manuals/obs-reference-guide/cha.obs.architecture.html#delta_store)
-to store changes in an incremental way independently from your chosen
-SCM.  The `obs_scm` service generates archives in this format, which
-is basically the `cpio` `newc` format.
+`obs_scm` is similar in concept to `tar_scm`, but instead of directly
+generating tarballs, it instead uses the new `obscpio` archive format
+(see below) as an intermediate space-efficient format in which to
+store the sources.
+
+**It is recommended to use `obs_scm` in favour to `tar_scm`**, because
+it helps to save a lot of disk space on the server side, especially
+when used in continuous integration (e.g. nightly builds and builds of
+pull requests).
+
+The usual source tarballs can be regenerated from this at build-time
+using the `tar` and `recompress` source services, so no changes to
+`.spec` files are required when switching to `obs_scm` and `obscpio`.
+Having said that, it may be more efficient to drop the build-time
+usage of `recompress`, since at build-time `rpmbuild` would decompress
+the same file soon after compressing it.  The decision whether to use
```

Comment:
well, this is only true when you also disable the run of local service or the same compression time would happen just before jumping into the build root

Reply:
Sorry, I'm too stupid to understand what you meant here, so I'm going to need some more help ;-)

The way I see it, the pipeline is:

SCM ---(obs_scm)---> `.obscpio` + `.obsinfo` 
    ---(tar)---> `.tar`
    ---(recompress)---> `.tar.xz`
    ---(rpmbuild)---> unpacked sources

Of these 4 steps, the last 3 have to happen at build-time, right?  And the first step (`obs_scm`) can either happen also at build-time (i.e. server-side), or client-side if it's set to disabled and manually triggered via `osc service disabledrun`.

This part of the README is just suggesting that the `recompress` part can be dropped, which would make `rpmbuild` quicker.  How that is related to the decision whether `obs_scm` is put in disabled mode or not?

Reply:
@adrianschroeter Any chance of a reply to this? ;-) Thanks!

Reply:
@aspiers

obs_scm cannot run at buildtime as it needs online access to fetch the sources from your repository.

Please @adrianschroeter correct me if I`m wrong

AFAIK - recompressing the sources makes building slower - indeed - but source rpms (srpm) should also get smaller - So I would recommend it - even if build is slower and more resources are temporarly used in OBS

Reply:
@M0ses wrote:
> obs_scm cannot run at buildtime as it needs online access to fetch the sources from your repository.

Ah, I think maybe I see the confusion now - I was using the term "build-time" in an imprecise way, to mean multiple different things:

- rpmbuild running in a buildroot in a VM
- source services running server-side prior to rpmbuild (I'm not sure exactly when server-side source services are triggered since I only ever used `mode="disabled"`, but I'm assuming `obs_scm` can run server-side just like `tar_scm` and all other source services - right?)
- when `osc service disabledrun` is manually run from the CLI prior to a client-side `osc build`

I guess really it should only be used to mean the first of these.

> AFAIK - recompressing the sources makes building slower - indeed - but source rpms (srpm) should also get smaller - So I would recommend it - even if build is slower and more resources are temporarly used in OBS

Actually it can be the opposite - double compression can make the srpms bigger!

http://mailman.suse.de/mlarch/SuSE/cloud-devel/2017/cloud-devel.2017.10/msg00090.html

Sorry to any non-SUSE employees reading for the internal-only link. This was also discussed on #opensuse-buildservice FreeNode IRC 2017/10/18 around 3pm (that's probably UTC, not sure) although I couldn't find any public logs of that.

Still confused by what @adrianschroeter meant in his comment though...

Reply:
right, obs_scm runs also server side. 

You are correct regarding disabling the recompression topic, how I think we should mention at least that you get very larger src.rpm then. Maybe suggesting that the use should mark it as NoSource in this situation.

Reply:
> you get very larger src.rpm then

Actually you don't:

```
2017-10-18 15:14:27     adrianS the only valid way is to say you don't care about src.rpms
2017-10-18 15:14:34     coolo   adrianS: they are compressed - but with gzip
2017-10-18 15:14:49     adrianS huh ... new to me ....
```

;-)  But sorry, I wrote some better text yesterday which gives more details around this, but forgot to git push it.  Doing that now ...

Reply:
k, just ignore me then ... sorry for confusion